xls/synthesis/timing_characterization_client_main.py

```diff
@@ -54,85 +88,304 @@ def _synth(stub: synthesis_service_pb2_grpc.SynthesisServiceStub,
+      result = _synth(stub, verilog_text, top_name)
+    except grpc.RpcError:
+      break
+
+    ps = 1e12/result.max_frequency_hz
+    result = delay_model_pb2.DataPoint()
+    result.operation.op = kop
+    result.operation.bit_count = bit_count
+    operand = result.operation.operands.add()
+    operand.bit_count = bit_count
+    result.delay = int(ps)
+    results.append(result)
+  model.data_points.extend(results)
+
+
+def _run_variadic_bitwise(
+    kop: str, model: delay_model_pb2.DelayModel,
+    stub: synthesis_service_pb2_grpc.SynthesisServiceStub) -> None:
+  # Add op_model to protobuf message
+  add_op_model = model.op_models.add(op=kop)
+  add_op_model.estimator.regression.factors.add(
+    source=delay_model_pb2.DelayFactor.OPERAND_BIT_COUNT,
+    operand_number=0)
+
+  op = ENUM2NAME_MAP[kop]
+
+  # Compute samples
+  widths = [1] + list(range(2, MAX_WIDTH, 2))
+  arity = list(range(2, 8))
+  combs = product(widths, arity)
+  results = []
+  for bit_count, arity in combs:
+    op_type = f'bits[{bit_count}]'
+    ir_text = op_module_generator.generate_ir_package(op, op_type, (op_type,) * arity)
+    module_name = f'{op}_{bit_count}_{arity}'
+    mod_generator_result = op_module_generator.generate_verilog_module(module_name, ir_text)
+    top_name = module_name + '_wrapper'
+    verilog_text = op_module_generator.generate_parallel_module([mod_generator_result], top_name)
+    try:
+      result = _synth(stub, verilog_text, top_name)
+    except grpc.RpcError:
+      break
+
+    ps = 1e12/result.max_frequency_hz
+    result = delay_model_pb2.DataPoint()
+    result.operation.op = kop
+    result.operation.bit_count = bit_count
+    operand = result.operation.operands.add()
+    operand.bit_count = bit_count
+    result.delay = int(ps)
+    results.append(result)
+  model.data_points.extend(results)
+
+
+def _run_arithmetic_unary(
+    kop: str, model: delay_model_pb2.DelayModel,
+    stub: synthesis_service_pb2_grpc.SynthesisServiceStub) -> None:
+  # Add op_model to protobuf message
+  add_op_model = model.op_models.add(op=kop)
+  add_op_model.estimator.regression.factors.add(
+    source=delay_model_pb2.DelayFactor.OPERAND_BIT_COUNT,
+    operand_number=0)
+
+  op = ENUM2NAME_MAP[kop]
+
+  results = []
+  for bit_count in [1] + list(range(2, MAX_WIDTH, 2)):
+    op_type = f'bits[{bit_count}]'
+    ir_text = op_module_generator.generate_ir_package(op, op_type, (op_type,))
+    module_name = f'{op}_{bit_count}'
+    mod_generator_result = op_module_generator.generate_verilog_module(module_name, ir_text)
+    top_name = module_name + '_wrapper'
+
+    verilog_text = op_module_generator.generate_parallel_module([mod_generator_result], top_name)
+    try:
+      result = _synth(stub, verilog_text, top_name)
+    except grpc.RpcError:
+      break
+
+    ps = 1e12/result.max_frequency_hz
+    result = delay_model_pb2.DataPoint()
+    result.operation.op = kop
+    result.operation.bit_count = bit_count
+    for i in range(2):
```

Comment:
I think this should only have one operand.

Reply:
(solved)